.atomist/editors/NewSpringBootRestService.ts

```diff
@@ -0,0 +1,170 @@
+/*
+ * Copyright © 2017 Atomist, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { PopulateProject } from '@atomist/rug/operations/ProjectGenerator'
+import { Project } from '@atomist/rug/model/Core'
+import { Pattern } from '@atomist/rug/operations/RugOperation'
+import { Generator, Parameter, Tags } from '@atomist/rug/operations/Decorators'
+import { PathExpression, PathExpressionEngine } from '@atomist/rug/tree/PathExpression'
+import { File } from '@atomist/rug/model/File'
+
+@Generator("NewSpringBootRestService", "create a new Spring Boot Rest Service project")
+@Tags("documentation", "java", "spring", "spring-boot", "spring-rest") 
+class NewSpringBootRestService implements PopulateProject {
+
+    // this is only necessary to avoid https://github.com/atomist/rug-resolver/issues/17
+    @Parameter({
+        displayName: "Project Name",
+        description: "name of project to be created",
+        pattern: Pattern.project_name,
+        validInput: "a valid GitHub project name consisting of alphanumeric, ., -, and _ characters",
+        minLength: 1,
+        maxLength: 100
+    })
+    project_name: string;
+
+    @Parameter({
+        displayName: "Maven Artifact ID",
+        description: "Maven artifact identifier, i.e., the name of the jar without the version, it is often the same as the project name",
+        pattern: "^[a-z][-a-z0-9_]*$", // Ideally this should be looking up artifact_id as a common pattern
```

Comment:
Isn't there a set pattern for this? Oh, perhaps this was the one that was missing that you created an issue for?